src/mesh/rectilineargrid.jl

```diff
@@ -20,36 +20,36 @@ julia> y = [0.0, 0.1, 0.3, 0.7, 0.9, 1.0]
 julia> RectilinearGrid(x, y)
 ```
 """
-struct RectilinearGrid{Datum,Dim,ℒ<:Len,V<:AbstractVector{ℒ}} <: Grid{𝔼{Dim},Cartesian{Datum,Dim,ℒ},Dim}
+struct RectilinearGrid{M<:Manifold,C<:CRS,Dim,V<:AbstractVector} <: Grid{M,C,Dim}
   xyz::NTuple{Dim,V}
   topology::GridTopology{Dim}
 
-  function RectilinearGrid{Datum}(
-    xyz::NTuple{Dim,<:AbstractVector{<:Len}},
-    topology::GridTopology{Dim}
-  ) where {Datum,Dim}
+  function RectilinearGrid{M,C}(xyz::NTuple{Dim,<:AbstractVector}, topology::GridTopology{Dim}) where {M,C,Dim}
     coords = float.(xyz)
     V = eltype(coords)
-    new{Datum,Dim,eltype(V),V}(coords, topology)
+    new{M,C,Dim,eltype(V),V}(coords, topology)
   end
 end
 
-RectilinearGrid{Datum}(xyz::NTuple{Dim,<:AbstractVector}, topology::GridTopology{Dim}) where {Datum,Dim} =
-  RectilinearGrid{Datum}(addunit.(xyz, u"m"), topology)
-
-function RectilinearGrid{Datum}(xyz::Tuple) where {Datum}
+function RectilinearGrid{M,C}(xyz::Tuple) where {M,C}
   coords = promote(collect.(xyz)...)
   topology = GridTopology(length.(coords) .- 1)
-  RectilinearGrid{Datum}(coords, topology)
+  RectilinearGrid{M,C}(coords, topology)
 end
 
-RectilinearGrid{Datum}(xyz...) where {Datum} = RectilinearGrid{Datum}(xyz)
+RectilinearGrid{M,C}(xyz...) where {M,C} = RectilinearGrid{M,C}(xyz)
+
+function RectilinearGrid(xyz::Tuple)
+  Dim = length(xyz)
+  C = Cartesian{NoDatum,Dim,Met{Float64}}
+  RectilinearGrid{𝔼{Dim},C}(args...)
+end
 
-RectilinearGrid(args...) = RectilinearGrid{NoDatum}(args...)
+RectilinearGrid(xyz...) = RectilinearGrid(xyz)
 
-vertex(g::RectilinearGrid{Datum}, ijk::Dims) where {Datum} = Point(Cartesian{Datum}(getindex.(g.xyz, ijk)))
+vertex(g::RectilinearGrid{M,C}, ijk::Dims) where {M,C} = Point(CoordRefSystems.reconstruct(C, getindex.(g.xyz, ijk)))
```

Comment:
let's break this into multiple lines, one defining `raw = getindex....` another doing reconstruction, and finally the point

Reply:
no need for M,C here either, you can crs(g) inside the reconstruct